### PR TITLE
DOC: Simplify axhspan example

### PR DESCRIPTION
**Pull request by timhoffm**
- Replace the clutter of many random plot elements by one meaningful example.
- Only cover spans. Remove the lines and instead cross-link to them.

### CODE REVIEW

**Comment by scottshambaugh** [comment]
`axvspan` isn't demoed anywhere else, and `axvline`, `axhline`, and `axline` are in other examples but just incidentally. I agree that the current example is cluttered, but maybe just reduce to one use of each function and rename the example to something more generic?

File: galleries/examples/subplots_axes_and_figures/axhspan_demo.py [hunk 1]
```diff
@@ -1,36 +1,38 @@
...
+often used to highlight data regions.
 """
 
 import matplotlib.pyplot as plt
 import numpy as np
 
-t = np.arange(-1, 2, .01)
-s = np.sin(2 * np.pi * t)
+np.random.seed(19680801)
+s = 2.9 * np.convolve(np.random.randn(500), np.ones(30) / 30, mode='valid')
```
[the remark this answers is not in the document]
Comment (timhoffm) [reply galleries/examples/subplots_axes_and_figures/axhspan_demo.py:15]: I'd like to have a "real use-case like" plot. The span must convey information. The example I'm alluding on is whether a singnal is in a fixed tolerance range. I'm not wedded to the particular plot. If you can come up with a better one, please take over. My boundary conditions are that (i) the span has to make sense in the plot ("why is it there") - IMHO spanning a sine does not fulfill this - and (ii) ideally, it looks like something that could potentially be found in the real world.

**Comment by timhoffm** [comment]
> Looks like this currently just demos `axhspan`. `axvspan` isn't demoed anywhere else, and `axvline`, `axhline`, and `axline` are in other examples but just incidentally. I agree that the current example is cluttered, but maybe just reduce to one use of each function and rename the example to something more generic?

A central aspect of the examples is discoverability via thumbnails:
image  image

The information you can put into that is limited. I believe it is not possible to cram  `axhspan`, `axvspan`, `axvline`, `axhline`, `axline` meaningfully into a single plot. In particular, I strongly favor to showcase functionality in a real-world like scenario. Just plotting a set of spans and lines without context is the mess of the current example (agreed that one could declutter it a bit, but it'd be still quite abstract). I therefore believe that separate examples for lines and spans are the way to go. This PR cross-links them to get the connection. As for "`axvspan` isn't demoed", I couldn't come up with a meaningful plot that needs both spans. I then rather have one good plot and just mention the other direction in text.

**Comment by timhoffm** [comment]
I've changed to two subplots, one horizontal and one vertical:
image

While two plots somewhat increase the code complexity, these two examples cover the use cases scope of spans better.

**Comment by timhoffm** [comment]
Merging on @story645's approval. Thanks!

**Pull request merged**